wordpress.org/public_html/wp-content/plugins/theme-directory/rest-api/class-theme-review-stats.php

```diff
@@ -0,0 +1,344 @@
+				'permission_callback' => array( $this, 'get_items_permissions' ),
+			)
+		);
+	}
+
+	/**
+	 * Get start date param from request.
+	 *
+	 * @returns string
+	 */
+	function get_start_date( $request ) {
+		$start_date = $request->get_param( 'startDate' );
+
+		if ( empty( $start_date ) ) {
+			$start_date = date( 'Y-m-d', strtotime( '-3 months' ) ); // Default to 3 months ago
+		}
+
+		return $start_date;
+	}
+
+	/**
+	 * Return whether the user can retrieve the data.
+	 *
+	 * @return bool
+	 */
+	function get_items_permissions( $request ) {
+		return current_user_can( 'edit_posts' );
+	}
+
+	/**
+	 * Returns an array of elements after the start date.
+	 */
+	function filter_items_before( $arr, $start_date ) {
+		return array_filter(
+			$arr,
+			function ( $row ) use ( $start_date ) {
+				return strtotime( $row[0] ) >= strtotime( $start_date );
+			}
+		);
+	}
+
+	/**
+	 * Returns a list averages review times by month.
+	 *
+	 * @return void
+	 */
+	public function get_average_review_days( $start_date ) {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		return $wpdb->get_results(
+			$wpdb->prepare(
+				"	
+				SELECT LEFT( upload_date, 7 ) as ym, AVG(days_to_review) as average_days_to_review
+
+				FROM (
+					SELECT p.post_name,
+					pm.meta_value,
+					substring_index( substring_index( pm.meta_value, '\"', 4 ), '\"', -1 ) AS upload_date,
+					p.post_date_gmt AS published_date,
+					DATEDIFF( p.post_date_gmt, substring_index( substring_index( pm.meta_value, '\"', 4 ), '\"', -1 ) ) as days_to_review
+			
+                    FROM {$wpdb->posts} p
+                    JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id AND pm.meta_key = '_upload_date'
+			
+					WHERE p.post_type = 'repopackage' and p.post_status = 'publish'
+			
+					ORDER BY p.ID desc
+				)a
+				
+				WHERE published_date > upload_date AND days_to_review IS NOT NULL AND upload_date >= %s
+
+				GROUP BY ym
+				ORDER BY upload_date ASC
+					",
+				$start_date
+			)
+		);
+	}
+
+	/**
+	 * Returns a list of published themes.
+	 *
+	 * @return void
+	 */
+	public function get_uploaded_themes() {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		return $wpdb->get_results(
+			"	
+			SELECT p.post_name as postName,
+				p.post_author as post_author,
+				substring_index( substring_index( pm.meta_value, '\"', 4 ), '\"', -1 ) AS author,
+				LEFT(p.post_date, 7) as published_on,
+				group_concat( t.slug SEPARATOR ', ' ) as tags
+
+			FROM {$wpdb->posts} p
+			JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id AND pm.meta_key = '_author'
+			LEFT JOIN {$wpdb->term_relationships} tr ON p.ID = tr.object_id
+			LEFT JOIN {$wpdb->term_taxonomy} tt ON tr.term_taxonomy_id = tt.term_taxonomy_id AND tt.taxonomy = 'post_tag'
+			LEFT JOIN {$wpdb->terms} t ON tt.term_id = t.term_id
+
+			WHERE p.post_type = 'repopackage' and p.post_status = 'publish'
+
+			Group by p.ID
+			Order By published_on asc
```

Comment:
nitpicking, it's best to keep SQLs as uppercase for SQL keywords & readability.
```suggestion
			WHERE p.post_type = 'repopackage' AND p.post_status = 'publish'

			GROUP BY p.ID
			ORDER BY published_on ASC
```